packages/ns-api/files/ns.devices

```diff
@@ -0,0 +1,1035 @@
+                'tx_packets': stat.get('stats64').get('tx').get('packets'),
+            }
+
+    # retrieve devices speed
+
+    for device in devices_to_show:
+        try:
+            with open(f'/sys/class/net/{device.get("name")}/speed') as f:
+                speed = int(f.read().strip())
+                device['speed'] = speed
+        except OSError:
+            # ignore read errors
+            pass
+
+    # sorted zones and devices
+
+    zones = [
+        {
+            'name': 'lan',
+            'devices': [],
+        },
+        {
+            'name': 'wan',
+            'devices': []
+        },
+        {
+            'name': 'guests',
+            'devices': []
+        },
+        {
+            'name': 'dmz',
+            'devices': []
+        },
+    ]
+
+    unassigned_zone = {
+        'name': 'unassigned',
+        'devices': []
+    }
+
+    fw_zones = get_all_by_type_as_list(uci, 'firewall', 'zone')
+
+    for device in devices_to_show:
+        iface_found = get_interface(device.get('name'), ifaces_from_config)
+
+        if not iface_found:
+            unassigned_zone['devices'].append(get_name(device))
+        else:
+            zone_found = get_firewall_zone(iface_found['.name'], fw_zones)
+
+            if not zone_found:
+                unassigned_zone['devices'].append(get_name(device))
+            else:
+                zone_obj = None
+
+                for z in zones:
+                    if z.get('name') == zone_found.get('name'):
+                        zone_obj = z
+                        break
+
+                if zone_obj:
+                    zone_obj['devices'].append(get_name(device))
+                else:
+                    zones.append({
+                        'name': zone_found.get('name'),
+                        'devices': [get_name(device)]
+                    })
+    zones.append(unassigned_zone)
+
+    # sort devices of every zone
+
+    for zone in zones:
+        zone['devices'].sort()
+
+    return {'devices_by_zone': zones, 'all_devices': all_devices}
+
+
+def get_firewall_zone(iface_name, fw_zones):
+    zone_found = None
+
+    for zone in fw_zones:
+        if zone.get('network') and iface_name in zone.get('network'):
+            zone_found = zone
+            break
+
+    return zone_found
+
+
+def get_name(device_or_iface):
+    if device_or_iface.get('.type') == 'interface':
+        return device_or_iface.get('.name')
+    else:
+        return device_or_iface.get('name')
+
+
+def get_interface(device_name, ifaces_from_config):
+    for iface in ifaces_from_config:
+        if iface.get('device') == device_name:
+            return iface
+
+
+def is_bridge(device):
+    if device.get('type') == 'bridge' or re.match(r"br[0-9]+", device.get('name')):
+        return True
+    else:
+        return False
+
+
+def should_omit_device_name(device_name):
+    return device_name == 'lo' or device_name.startswith('ifb-')
```

Comment:
Probably we should also skip all tun devices:
```
```suggestion
    return device_name == 'lo' or device_name.startswith('ifb-') or device_name.startswith('tun')
```
```